test/cmd/basicresources.sh

```diff
@@ -6,102 +6,105 @@ set -o pipefail
 # Test that external services are exposable
-oc create -f test/fixtures/external-service.yaml
-oc expose svc/external
-[ "$(oc get route external | grep 'external=service')" ]
-oc delete route external
-oc delete svc external
+os::cmd::expect_success 'oc create -f test/fixtures/external-service.yaml'
+os::cmd::expect_success 'oc expose svc/external'
+os::cmd::expect_success_and_text 'oc get route external' 'external=service'
+os::cmd::expect_success 'oc delete route external'
+os::cmd::expect_success 'oc delete svc external'
 # Expose multiport service and verify we set a port in the route
-oc create -f test/fixtures/multiport-service.yaml
-oc expose svc/frontend --name route-with-set-port
-os::util::get_object_assert 'route route-with-set-port' "{{.spec.port.targetPort}}" '8080'
+os::cmd::expect_success 'oc create -f test/fixtures/multiport-service.yaml'
+os::cmd::expect_success 'oc expose svc/frontend --name route-with-set-port'
+os::cmd::expect_success "os::util::get_object_assert 'route route-with-set-port' '{{.spec.port.targetPort}}' '8080'"
 echo "expose: ok"
 
-oc delete all --all
+os::cmd::expect_success 'oc delete all --all'
 
 # switch to test user to be sure that default project admin policy works properly
-oc policy add-role-to-user admin test-user
-oc login -u test-user -p anything
+os::cmd::expect_success 'oc policy add-role-to-user admin test-user'
+os::cmd::expect_success 'oc login -u test-user -p anything'
 
-oc run --image=openshift/hello-openshift test
-oc run --image=openshift/hello-openshift --generator=run-controller/v1 test2
-oc run --image=openshift/hello-openshift --restart=Never test3
-oc delete dc/test rc/test2 pod/test3
+os::cmd::expect_success 'oc run --image=openshift/hello-openshift test'
+os::cmd::expect_success 'oc run --image=openshift/hello-openshift --generator=run-controller/v1 test2'
+os::cmd::expect_success 'oc run --image=openshift/hello-openshift --restart=Never test3'
+os::cmd::expect_success 'oc delete dc/test rc/test2 pod/test3'
 
-oc process -f examples/sample-app/application-template-stibuild.json -l name=mytemplate | oc create -f -
-oc delete all -l name=mytemplate
-oc new-app https://github.com/openshift/ruby-hello-world
-[ "$(oc get dc/ruby-hello-world)" ]
+os::cmd::expect_success 'oc process -f examples/sample-app/application-template-stibuild.json -l name=mytemplate | oc create -f -'
+os::cmd::expect_success 'oc delete all -l name=mytemplate'
+os::cmd::expect_success 'oc new-app https://github.com/openshift/ruby-hello-world'
+os::cmd::expect_success 'oc get dc/ruby-hello-world'
 
-oc get dc/ruby-hello-world --template='{{ .spec.replicas }}' | grep -q 1
-oc patch dc/ruby-hello-world -p '{"spec": {"replicas": 2}}'
-oc get dc/ruby-hello-world --template='{{ .spec.replicas }}' | grep -q 2
+os::cmd::expect_success_and_text "oc get dc/ruby-hello-world --template='{{ .spec.replicas }}'" '1'
+os::cmd::expect_success "oc patch dc/ruby-hello-world -p '{\"spec\": {\"replicas\": 2}}'"
```

Comment:
This isn't the best but if you want both literal single-quotes and double-quotes in a string this is the only way to do it on the same line. We could also declare the patch text as a variable and pass that in.